api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java

```diff
@@ -740,7 +750,262 @@ public void shouldTestPurgeProgramAttributeType() throws Exception {
                 int totalAttributeTypes = pws.getAllProgramAttributeTypes().size();
 		pws.purgeProgramAttributeType(programAttributeType);
 		assertEquals((totalAttributeTypes - 1), pws.getAllProgramAttributeTypes().size());
-}
+	}
+	
+	
+	@Test
+	public void getPrograms_shouldTestGetPrograms() {
+		List<Program> malPrograms = pws.getPrograms("MAL");
+		List<Program> prPrograms = pws.getPrograms("PR");
+		assertEquals(malPrograms.size(), 1);
+		assertEquals(prPrograms.size(), 3);
+	}
+	
+	@Test
+	public void retireProgram_shouldTestRetireProgram() {
+		/*This method should retire a program, by setting a reason, and also
+		setting it's workflows and workflow states to retired*/
+		Program programTest = pws.getAllPrograms().get(0);
+		Program program = pws.retireProgram(programTest, "expired");
+		for (ProgramWorkflow workflow : program.getWorkflows()) {
+			assertEquals(workflow.getRetired(), true);
+			for (ProgramWorkflowState state : workflow.getStates()) {
+				assertEquals(state.getRetired(), true);
+				assertNotNull(workflow.getStates());
```

Comment:
You should move this line before the `for` loop.

Reply:
Did this

Reply:
@alinmihaila99 It looks like there are a few of the comments where you said you made the changes, but they aren't reflected in the current PR. Could you make sure this is the most up-to-date code?